lib/socket.ts

```diff
@@ -6,6 +6,14 @@ import { Server } from "./server";
 
 const debug = debugModule("engine:socket");
 
+type PacketType = "error" | "message" | "open" | "ping" | "pong";
+
+export interface Packet {
+  type: PacketType;
+  options: { compress: boolean };
+  data?: any;
```

Comment:
@darrachequesne  I noticed within the engine codebase `string` is sufficient, but IRL i also got array type, can this really be anything?